### ISSUE TITLE
Sort members isn't stable

### ISSUE THREAD

**Issue by bwilkerson** [dart-lang/sdk#44759]
The 'sort members' operation should not make changes except when necessary because of moving a member to the correct location, but it appears to not be stable when some members have a blank line between them and others don't. See https://dart-review.googlesource.com/c/sdk/+/180721/2/pkg/analyzer/lib/src/dart/micro/analysis_context.dart#107 for an example.

**Comment by bwilkerson** [comment]
@scheglov

**Comment by incendial** [comment]
@bwilkerson @srawlins is this issue still valid?

I'd like to share the implementation I have for DCM where this problem is solved. Can create a CL. WDYT?

**Comment by bwilkerson** [comment]
To the best of my knowledge this issue hasn't been addressed.

Contributions are always welcome.

**Comment by incendial** [comment]
Done: https://dart-review.googlesource.com/c/sdk/+/281441

**Comment by incendial** [comment]
@srawlins curious why the failed test is the one I've not touched. You're waiting for me to fix it before the review?

**Comment by srawlins** [comment]
Yes please. I imagine this test failed because you edited the sorting code.

**Comment by incendial** [comment]
> Yes please. I imagine this test failed because you edited the sorting code.

Ah, didn't think of that. Fixed (was a bug, actually).

**Comment by incendial** [comment]
@srawlins the more I think about this CL, the more I wonder - maybe instead of the hardcoded `' '` it should be calculated (if the user has tabs instead of spaces)?

**Comment by incendial** [comment]
@srawlins @bwilkerson any update on the review?

**Comment by srawlins** [comment]
No update, it could take a while, apologies.

**Comment by incendial** [comment]
@srawlins should I keep an eye on this and ping from time to time or you'll get back when you have time?

**Comment by srawlins** [comment]
Sorry for the delayed response!! I'm just kicked off CI again on your PR and I'd love to merge it soon.

**Comment by incendial** [comment]
@srawlins I see that checks failed because of the analysis, but this looks fast to fix. Will there be more review comments, so I can address them and the failures at once?

**Comment by srawlins** [comment]
A fair question! As it has not been '+1''d yet. I reviewed it and I think I will approve ASAP. Let me take one more look.

**Comment by FMorschel** [comment]
I'm not sure this is still happening. I could not replicate it. Can anyone confirm this?
